spec/mongoid/geospatial/fields/polygon_spec.rb

```diff
@@ -38,6 +38,12 @@
       expect(geom.radius_sphere(10)[1]).to be_within(0.001).of(0.001569)
     end
 
+    it 'should not fail to demongoize BSON documents' do
+      expect do
+        Mongoid::Geospatial::Polygon.demongoize(BSON::Document.new)
```

Comment:
Either way I think @cschell you could really help by writing a failing test case for the actual scenario, even if the fix only works coincidentally (aka this is not the best or the right fix).